### PR TITLE
Avro article - Part 1

### PR DESCRIPTION
**Pull request by antonmry**

### CODE REVIEW

File: src/jbake/content/blog/2020/20200813-Gentle-and-practical-introduction-to-Apache-Avro-Part-1.md [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+The way to add the ID to the message is also simple: one byte with the value `0`
+in the case of Confluent, 4 bytes with the ID and the rest of the data. It's
+documented in the [Wire Format] entry.
+
+## Environment setup
+
+Using the Confluent Avro serializer/deserializer, the process is quite
+straight-forward. Let's try it using the Confluent Community Docker version.
+The setup is documented in the [Quick Start for Apache Kafka using Confluent
+Platform Community Components (Docker)] which it's summarized here:
```
Comment (dariocazas) [inline src/jbake/content/blog/2020/20200813-Gentle-and-practical-introduction-to-Apache-Avro-Part-1.md:58]: Break link. Perhaps to https://docs.confluent.io/platform/current/quickstart/cos-docker-quickstart.html

File: src/jbake/content/blog/2020/20200813-Gentle-and-practical-introduction-to-Apache-Avro-Part-1.md [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+
+The output should be:
+
+> Created topic test.` message.
+
+To test it, we are going to create a Kafka Producer and a Kafka Consumer.
+
+## Kafka Producer with Confluent Schema Registry
+
+Download the [kafka-java-client-examples] project and open it with your
```
Comment (dariocazas) [inline src/jbake/content/blog/2020/20200813-Gentle-and-practical-introduction-to-Apache-Avro-Part-1.md:85]: Break link, perhaps to https://github.com/antonmry/kafka-java-client-examples

File: src/jbake/content/blog/2020/20200813-Gentle-and-practical-introduction-to-Apache-Avro-Part-1.md (hunk 1 above)
Comment (dariocazas) [reply src/jbake/content/blog/2020/20200813-Gentle-and-practical-introduction-to-Apache-Avro-Part-1.md:58]: Ok, I see it... Ignore this comment ;)

File: src/jbake/content/blog/2020/20200813-Gentle-and-practical-introduction-to-Apache-Avro-Part-1.md [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+
+```sh
+curl -X GET http://localhost:8081/config
+```
+
+By default, it should return:
+
+> {"compatibilityLevel":"BACKWARD"}
+
+Backward compatibility means we could delete a new field to our schema so new
```
Comment (dariocazas) [inline src/jbake/content/blog/2020/20200813-Gentle-and-practical-introduction-to-Apache-Avro-Part-1.md:197]: Complex to understand... 

Backward compatibility means we could evolve the schema from the producer side, and the consumers don't break. For example:

- The evolution adds a new field:  the consumer will ignore it, is backward compatibility.
- The evolution delete old fields: ok, check it... Does the previous schema define a default value for the field? Then the consumer assumes this default value, the deletion is backward compatibility. If the previous schema doesn't have a default value for the field, no backward compatibility is guaranteed.

We could test it using (...)

File: src/jbake/content/blog/2020/20200813-Gentle-and-practical-introduction-to-Apache-Avro-Part-1.md [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+the Producer deleting one field:
+
+```json
+{
+ "namespace": "com.galiglobal.examples.testavro",
+ "type": "record",
+ "name": "Test",
+ "fields": [
+     {"name": "id", "type": "string"}
+ ]
```
Comment (dariocazas) [inline src/jbake/content/blog/2020/20200813-Gentle-and-practical-introduction-to-Apache-Avro-Part-1.md:210]: This field deletion is backward compatible with the initial JSON?

Comment (antonmry) [reply src/jbake/content/blog/2020/20200813-Gentle-and-practical-introduction-to-Apache-Avro-Part-1.md:214]: yes, as discussed. I've updated the example to make it more clear

File: src/jbake/content/blog/2020/20200813-Gentle-and-practical-introduction-to-Apache-Avro-Part-1.md (hunk 3 above)
Comment (antonmry) [reply src/jbake/content/blog/2020/20200813-Gentle-and-practical-introduction-to-Apache-Avro-Part-1.md]: changed. It should be more clear now. Thanks for the feedback

File: src/jbake/content/blog/2020/20200813-Gentle-and-practical-introduction-to-Apache-Avro-Part-1.md (hunk 2 above)
Comment (antonmry) [reply src/jbake/content/blog/2020/20200813-Gentle-and-practical-introduction-to-Apache-Avro-Part-1.md:85]: Fixed! Thanks

**Pull request merged**